### PR TITLE
(bug) Removing a user having multiple role from owner tab also remove the other roles associated to that user

### PR DESCRIPTION
**Pull request by saxo-lalrishav**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] https://github.com/linkedin/datahub/issues/2567
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-web-react/src/app/entity/shared/Ownership.tsx [hunk 1]
```diff
@@ -105,13 +105,14 @@ export const Ownership: React.FC<Props> = ({ owners, lastModifiedAt, updateOwner
         setStagedOwners(newStagedOwners);
     };
 
-    const onDelete = (urn: string) => {
+    const onDelete = (urn: string, type: OwnershipType) => {
         const updatedOwners = owners
-            .filter((owner) => !(owner.owner.urn === urn))
+            .filter((owner) => !(owner.owner.urn === urn && owner.type === type))
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/shared/Ownership.tsx:110]: is it possible for an two owners to have the same urn but not have the same type?

**Review by gabe-lyons — COMMENTED** [review]
This looks good to me, but I think the major source of this issue is that we do not currently do validation on owner writes. It seems like you added the same urn as a group and as a user in order to reproduce this bug- we should make sure that isn't possible in the future.

File: datahub-web-react/src/app/entity/shared/Ownership.tsx (hunk 1 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/shared/Ownership.tsx:110]: Sorry, i think i need to filter it with role and not with type
Right ?

Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/shared/Ownership.tsx:110]: Because a user can have multiple role there

Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/shared/Ownership.tsx:110]: So it should be ``` owner.type === role ```

Comment (gabe-lyons) [reply datahub-web-react/src/app/entity/shared/Ownership.tsx:110]: Ah I see- I think that may be the solution.

**Review by gabe-lyons — APPROVED** [review]
looks good to me- thanks @saxo-lalrishav

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**